packages/commonwealth/server/routes/searchProfiles.ts

```diff
@@ -0,0 +1,89 @@
+import { QueryTypes } from 'sequelize';
+import { ALL_CHAINS } from '../middleware/databaseValidationService';
+import { AppError } from '../../../common-common/src/errors';
+import type { DB } from '../models';
+
+const MIN_SEARCH_QUERY_LENGTH = 3;
+
+export const Errors = {
+  InvalidChain: 'Invalid chain',
+  QueryMissing: 'Must enter query to begin searching',
+  QueryTooShort: 'Query must be at least 4 characters',
+  NoChains: 'No chains resolved to execute search',
+};
+
+type SearchProfilesQuery = {
+  search?: string;
+  chain?: string;
+};
+
+const searchProfiles = async (models: DB, req, res) => {
+  const options = req.query as SearchProfilesQuery;
```

Comment:
Feel free to use `TypedRequestQuery` and `TypedResponse` to get easier typechecking here.